service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/EhrResendControllerTest.java

```diff
@@ -0,0 +1,269 @@
+                                                           .received(FIVE_DAYS_AGO)
+                                                           .rootId(generateRandomUppercaseUUID())
+                                                           .build());
+        ehrExtractStatus.setEhrRequest(EhrExtractStatus.EhrRequest.builder().nhsNumber(NHS_NUMBER).build());
+        ehrExtractStatus.setEhrExtractCorePending(EhrExtractStatus.EhrExtractCorePending.builder().build());
+        ehrExtractStatus.setEhrContinue(EhrExtractStatus.EhrContinue.builder().build());
+        ehrExtractStatus.setGpcAccessDocument(EhrExtractStatus.GpcAccessDocument.builder().build());
+
+        doReturn(Optional.of(ehrExtractStatus)).when(ehrExtractStatusRepository).findByConversationId(CONVERSATION_ID);
+
+        Instant now = Instant.now().truncatedTo(ChronoUnit.MILLIS);
+        doReturn(now).when(timestampService).now();
+
+        ehrResendController.scheduleEhrExtractResend(CONVERSATION_ID);
+
+        var updatedEhrExtractStatus = ehrExtractStatusRepository.findByConversationId(ehrExtractStatus.getConversationId());
+        var taskDefinition = GetGpcStructuredTaskDefinition.getGetGpcStructuredTaskDefinition(randomIdGeneratorService, ehrExtractStatus);
+
+        verify(taskDispatcher, times(1)).createTask(taskDefinition);
+        assertEquals(now, updatedEhrExtractStatus.get().getMessageTimestamp());
+        assertEquals(now, updatedEhrExtractStatus.get().getCreated());
+        assertEquals(now, updatedEhrExtractStatus.get().getUpdatedAt());
+        assertNull(updatedEhrExtractStatus.get().getEhrExtractCorePending());
+        assertNull(updatedEhrExtractStatus.get().getEhrContinue());
+        assertNull(updatedEhrExtractStatus.get().getAckPending());
+        assertNull(updatedEhrExtractStatus.get().getEhrReceivedAcknowledgement());
+        assertNull(updatedEhrExtractStatus.get().getGpcAccessDocument());
+    }
+
+    @Test
+    void When_AnEhrExtractHasNotFailedAndAnotherResendRequestArrives_Expect_FailedOperationOutcome() throws JsonProcessingException {
+
+        var details = new CodeableConcept();
+        var codeableConceptCoding = new Coding();
+        codeableConceptCoding.setSystem(GPCONNECT_ERROR_OR_WARNING_CODE);
+        codeableConceptCoding.setCode(INTERNAL_SERVER_ERROR);
+        details.setCoding(List.of(codeableConceptCoding));
+
+        final EhrExtractStatus IN_PROGRESS_EXTRACT_STATUS = EhrExtractStatus.builder()
+            .conversationId(CONVERSATION_ID)
+            .ackPending(EhrExtractStatus.AckPending.builder().typeCode("AA").build())
+            .ackToRequester(EhrExtractStatus.AckToRequester.builder().typeCode("AA").build())
+            .ehrRequest(EhrExtractStatus.EhrRequest.builder().nhsNumber(NHS_NUMBER).toAsid(TO_ASID_CODE).fromAsid(FROM_ASID_CODE).build())
+            .build();
+
+        doReturn(Optional.of(IN_PROGRESS_EXTRACT_STATUS)).when(ehrExtractStatusRepository).findByConversationId(CONVERSATION_ID);
```

Comment:
and again, shouldn't be using doReturn()

Reply:
ok, will change it